api/src/main/scala/org/ensime/api/common.scala

```diff
@@ -129,10 +131,80 @@ object OffsetRange extends ((Int, Int) => OffsetRange) {
 // such as files/dirs, existance, content hints
 // (java/scala/class/resource) in the type, validated at construction
 // (and can be revalidated at any time)
-sealed trait EnsimeFile
-final case class RawFile(file: Path) extends EnsimeFile
+sealed trait EnsimeFile {
+  def path: Path
```

Comment:
in fact, no code in `api` is the rule. This remains a clean data structure as it is exported and viewed across languages. It is also Apache2 licensed whereas the code you copied is GPLv3.